components/projects/EnvironmentVariableEditor.tsx

```diff
@@ -0,0 +1,192 @@
+      className={`${
+        isDragActive ? "border-teal-300" : "border-darker"
+      } mt-4 w-full border-2 transition duration-300`}
+    >
+      {envKeys.length > 0 ? (
+        <div className="py-16">
+          {envKeys?.map((envPair, index) => (
+            <div
+              key={new Date().toString() + index}
+              className="flex items-center gap-8 px-4"
+            >
+              <CustomInput
+                name={envPair.envKey}
+                onFocus={() => (pastingInputIndex.current = index)}
+                type="text"
+                defaultValue={envPair.envKey}
+                className="my-1 w-full font-mono"
+                onPaste={handlePaste}
+                placeholder="eg. CLIENT_ID"
+              />
+
+              <div className="my-1 flex w-full items-center space-x-2">
+                <InputGroup
+                  full
+                  icon={<EyeIcon className="h-4 w-4" />}
+                  name={envPair.envValue}
+                  type="password"
+                  autoComplete="off"
+                  defaultValue={envPair.envValue}
+                  className="font-mono"
+                />
+
+                <Button
+                  onClick={() => handleRemoveEnvPairClick(index)}
+                  className="rounded"
+                  outline
+                  small
+                >
+                  <XIcon className="h-4 w-4" />
+                </Button>
+              </div>
+            </div>
+          ))}
+
+          <div className="mt-4 px-4">
+            <Button small outline onClick={() => handleAddMoreEnvClick()}>
+              Add more
+            </Button>
+          </div>
+        </div>
+      ) : (
+        <div {...getRootProps()} className="py-16 text-center">
+          <DragDropIcon className="mx-auto h-12 w-12" />
+          <h3 className="mt-2 text-xl text-gray-400">
+            Drag and drop .env files
+          </h3>
+          <input
+            {...getInputProps()}
+            type="file"
+            className="hidden"
+            accept="env"
+          />
+          <p className="mx-auto mt-1 max-w-md text-sm text-gray-200 text-lighter">
+            You can also{" "}
+            <span
+              onClick={open}
+              className="text-teal-300 transition duration-300 hover:cursor-pointer hover:underline"
+            >
+              click here
+            </span>{" "}
+            to import, copy/paste contents in .env file, or create{" "}
+            <span
+              onClick={handleAddMoreEnvClick}
+              className="text-teal-300 transition duration-300 hover:cursor-pointer hover:underline"
+            >
+              one at a time.
+            </span>
+          </p>
+        </div>
+      )}
+    </Container>
+  );
+}
+
+function CustomInput({
+  disabled,
+  className,
+  ...props
+}: ComponentProps<"input">) {
+  return (
+    <input
+      {...props}
+      disabled={disabled}
+      className={`${
```

Comment:
Lets use `clsx` for these complicated classNames

Reply:
done